omgitsjan/DiscordBotAI
# [FEATURE] Create HttpService with a general HttpRequest method

**Pull Request Description**
This pull request addresses issue #[ISSUE_NUMBER] by introducing a new HttpService class, which centralizes and simplifies the process of making HTTP requests throughout the application. The primary feature of this service is the addition of a general-purpose HttpRequest method that supports various types of HTTP requests, including GET, POST, PUT, DELETE, and more.

**Key Changes**
Creation of the HttpService class with the HttpRequest method: This new class provides a general-purpose method for making HTTP requests, making it easier for developers to work with different types of requests.

Refactoring of existing code that makes HTTP requests: To improve consistency and maintainability, all existing code making HTTP requests has been refactored to use the new HttpRequest method provided by the HttpService class.

Updating unit tests and adding new tests: Existing unit tests have been updated, and new tests have been added to ensure that the HttpRequest method works correctly and covers various scenarios.

**Benefits**
The implementation of this HttpService aims to improve code maintainability and reduce the complexity of making HTTP requests in different parts of the application. By using a centralized approach for handling HTTP requests, developers can focus on implementing business logic and improving overall code quality.

**Steps to Test**
Run the updated unit tests to confirm that the HttpRequest method works as expected.
Manually test various use cases and scenarios in the application to ensure that the refactored code utilizing the new HelperService works correctly.
